pyvrp/cpp/ProblemData.cpp

```diff
@@ -72,6 +72,21 @@ ProblemData::ProblemData(std::vector<Client> const &clients,
                                        return sum + type.numAvailable;
                                    }))
 {
+
+    if (clients.size() == 0)  // at least one client (the depot) is required
+        throw std::invalid_argument("Clients must not be empty.");
+
+    auto const &depot = clients[0];
+
+    if (depot.demand != 0)
+        throw std::invalid_argument("Depot demand must be 0.");
+
+    if (depot.serviceDuration != 0)
+        throw std::invalid_argument("Depot service duration must be 0.");
+
+    if (depot.releaseTime != 0)
+        throw std::invalid_argument("Depot release time must be 0.");
```

Comment:
I find it odd that we can only check depot data here. Suggestions:
- Rename `Client` to `Node` and add an `is_depot` flag to the `Client` constructor. We can then put these checks in the `Node` constructor.
- Introduce a new `Depot` class that inherits from `Client`.

This also depends on #305 and @N-Wouda plans for pickup-and-delivery.

Reply:
I'm leaning the second, and storing depots in a separate list. They probably won't be in a class hierarchy; I'll spec out some concepts for this. But I want to try some options first before settling on any one approach.